internal/xds/translator/translator.go

```diff
@@ -577,3 +600,82 @@ func addXdsCluster(tCtx *types.ResourceVersionTable, args *xdsClusterArgs) error
 	}
 	return nil
 }
+
+const (
+	DefaultEndpointType EndpointType = iota
+	Static
+	EDS
+)
+
+func buildXdsUpstreamTLSCertSecret(tlsConfig ir.TLSBundle) *tlsv3.Secret {
+	// Build the tls secret
+
+	return &tlsv3.Secret{
+		Name: tlsConfig.GetXdsCertSecretName(), // GetBackendTLSCertSecretName(tlsConfig),
+		Type: &tlsv3.Secret_TlsCertificate{
+			TlsCertificate: &tlsv3.TlsCertificate{
+				CertificateChain: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.CertificateByte},
+				},
+				PrivateKey: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.PrivateKeyByte},
+				},
+			},
+		},
+	}
+}
+
+func buildXdsUpstreamTLSCASecret(tlsConfig ir.TLSBundle) *tlsv3.Secret {
+	// Build the tls secret
+
+	return &tlsv3.Secret{
+		Name: tlsConfig.GetXdsCaSecretName(), // GetBackendTLSCaSecretName(tlsConfig),
+		Type: &tlsv3.Secret_ValidationContext{
+			ValidationContext: &tlsv3.CertificateValidationContext{
+				TrustedCa: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.CaCertificate},
+				},
+			},
+		},
+	}
+}
+
+func buildXdsUpstreamTLSSocketWthCert(tlsConfigs *ir.TLSBundle) (*corev3.TransportSocket, error) {
+
+	tlsCtx := &tlsv3.UpstreamTlsContext{
+		CommonTlsContext: &tlsv3.CommonTlsContext{
+			TlsCertificateSdsSecretConfigs: func() []*tlsv3.SdsSecretConfig {
+				if !tlsConfigs.Mtls() {
+					return nil
+				}
+				x := []*tlsv3.SdsSecretConfig{
+					{
+						Name:      tlsConfigs.GetXdsCertSecretName(),
+						SdsConfig: makeConfigSource(),
+					},
+				}
+				return x
+			}(),
+			ValidationContextType: &tlsv3.CommonTlsContext_ValidationContextSdsSecretConfig{
+				ValidationContextSdsSecretConfig: &tlsv3.SdsSecretConfig{
+					Name:      tlsConfigs.GetXdsCaSecretName(),
+					SdsConfig: makeConfigSource(),
+				},
+			},
+		},
+		Sni:                tlsConfigs.Hostname,
+		AllowRenegotiation: true,
+	}
+
+	tlsCtxAny, err := anypb.New(tlsCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev3.TransportSocket{
+		Name: wellknown.TransportSocketTLS, // "envoy.transport_sockets.starttls",
```

Comment:
is it `starttls` or plain `tls`?

Reply:
TransportSocketTLS = "envoy.transport_sockets.tls"

Reply:
needs a change in comment

Reply:
fixed